compiler/circle-execution-plan/src/ExecutionOrderEstimator.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ExecutionOrderEstimator.h"
+
+namespace circle_planner
+{
+namespace
+{
+
+class Frame final
+{
+public:
+  Frame(loco::Node *ptr) : _ptr{ptr}, _pos{-1}
+  {
+    // DO NOTHING
+  }
+
+public:
+  loco::Node *ptr(void) const { return _ptr; }
+  int64_t pos(void) const { return _pos; }
+
+  void advance(void) { _pos += 1; }
+
+private:
+  loco::Node *_ptr = nullptr;
+  int64_t _pos = -1;
+};
+
+bool isExecutableNode(const luci::CircleNode *node)
+{
+  switch (node->opcode())
+  {
+      // These nodes denote inputs / outputs of a graph.
+    case luci::CircleOpcode::CIRCLECONST:
+    case luci::CircleOpcode::CIRCLEINPUT:
+    case luci::CircleOpcode::CIRCLEOUTPUT:
+    case luci::CircleOpcode::CIRCLEOUTPUTEXCLUDE:
+      // The following nodes denote outputs of multiple-output nodes.
+    case luci::CircleOpcode::CIRCLEIFOUT:
+    case luci::CircleOpcode::CIRCLESPLITOUT:
+    case luci::CircleOpcode::CIRCLEUNPACKOUT:
+    case luci::CircleOpcode::CIRCLEWHILEOUT:
+      return false;
+    default:
+      return true;
+  }
+}
+
+bool NextSet(std::vector<uint32_t> &vector)
+{
+  int n = vector.size();
```

Comment:
In case we try to iterate over empty vector
```suggestion
  int n = vector.size();
  if (n == 0)
    return false;
```